geoh5py/objects/surveys/electromagnetics/base.py

```diff
@@ -235,102 +234,31 @@ def components(self) -> dict | None:
 
         return None
 
-    def copy(  # pylint: disable=too-many-arguments
-        self,
-        parent: Group | Workspace | None = None,
-        *,
-        copy_children: bool = True,
-        clear_cache: bool = False,
-        mask: np.ndarray | None = None,
-        cell_mask: np.ndarray | None = None,
-        copy_complement: bool = True,
-        **kwargs,
-    ):
-        """
-        Sub-class extension of :func:`~geoh5py.objects.cell_object.CellObject.copy`.
-        """
-        if parent is None:
-            parent = self.parent
-
-        new_entity = super().copy(
-            parent=parent,
-            clear_cache=clear_cache,
-            copy_children=copy_children,
-            mask=mask,
-            cell_mask=cell_mask,
-            omit_list=OMIT_LIST,
-            **kwargs,
-        )
-
-        # Copy metadata except reference to entities UUID
-        for key, value in self.metadata["EM Dataset"].items():
-            if not isinstance(value, (uuid.UUID, type(None))):
-                new_entity.edit_em_metadata({key: value})
-
-        if copy_complement:
-            self.copy_complement(
-                new_entity,
-                parent=parent,
-                copy_children=copy_children,
-                clear_cache=clear_cache,
-                mask=mask,
-            )
-
-        return new_entity
-
-    def copy_complement(
-        self,
-        new_entity,
-        *,
-        parent: Group | Workspace | None = None,
-        copy_children: bool = True,
-        clear_cache: bool = False,
-        mask: np.ndarray | None = None,
-    ) -> BaseEMSurvey | None:
+    def get_complement_mask(
+        self, mask: np.ndarray | None, complement: BaseEMSurvey
+    ) -> np.ndarray:
         """
-        Copy the complement entity to the new entity.
+        Get the complement mask based on the input mask.
 
-        :param new_entity: New entity to copy the complement to.
-        :param parent: Parent group or workspace.
-        :param copy_children: Copy children entities.
-        :param clear_cache: Clear the cache.
-        :param mask: Mask on vertices to apply to the data.
+        :param mask: Mask on the vertices.
         """
```

Comment:
Should we have checkers for the inputs of our API?
Could be good no?

Reply:
yes